### DISCUSSION TITLE
mplhep logo placement is slow..

### DISCUSSION THREAD

**Discussion by MoAly98** [scikit-hep/mplhep#382] [General] [answered]
Hi,

We are working on a framework that is looking to utilise mplhep to plot a large number of histograms. One use case will involve plotting O(100) histograms. It seems like using the following line of code:

```
text = hep.atlas.label(ax=ax, label=logotext, data=data, lumi=lumi)
```

causes the run-time for one plot to increase by O(3000) times with respect to a simple python `ax.text()` call. It takes ~1.8s to run the above line, whereas running the following:

```
ax.text(x=0.05,y=0.95, s = "ATLAS")
```

takes 520 microseconds. Is this expected behaviour of the logo placement? 

The profilingwas done by `lprun` magic on jupyter:
With using `maptlolib`` ax.text`:

```
Timer unit: 1e-06 s
Total time: 0.169388 s

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   643                                                   
   644                                                               # add atlas logo and text
   645         1        520.0    520.0      0.3          self.set_explabel(ax)
```
With using `mplhep`` hep.atlas.label`:

```
Timer unit: 1e-06 s
Total time: 1.97478 s

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   642                                                   
   643                                                                       # add atlas logo and text
   644         1    1804167.0 1804167.0     91.4          self.set_explabel(ax)
   645                                                           

```

**Comment by andrzejnovak** [comment, marked as answer]
Hi @MoAly98, thanks for the report. There's already an issue #338, though I haven't had a chance to fix it yet. The reason why it's slow is that  the figure is drawn under the hood to find the right position for the second part of the label, this could presumably be estimated instead if calling the draw.

**Reply by andrzejnovak** [reply]
@MoAly98 Can you check if the issue improves in `mplhep 0.3.24`
